### PR TITLE
Fix threading for python gui

### PR DESCRIPTION
**Pull request by moon-jam**

### CODE REVIEW

File: python/inputmodule/gui/ledmatrix.py [hunk 1]
```diff
@@ -37,15 +41,17 @@ def countdown(dev, seconds):
...
+        breathing(dev)
+        # blinking(dev)
 
 
 def blinking(dev):
     """Blink brightness high/off every second.
     Keeps currently displayed grid"""
-    while True:
+    set_status('blinking')
+    while True and get_status() == 'blinking':
```
Comment (spenserblack) [inline python/inputmodule/gui/ledmatrix.py:54]: Nitpick: `True and` is redundant boolean logic
```suggestion
    while get_status() == 'blinking':
```

File: python/inputmodule/gui/ledmatrix.py [hunk 2]
```diff
@@ -57,7 +63,9 @@ def blinking(dev):
 
 def random_eq(dev):
     """Display an equlizer looking animation with random values."""
-    while True:
+    animate(dev, False)
+    set_status('random_eq')
+    while True and get_status() == 'random_eq':
```
Comment (spenserblack) [inline python/inputmodule/gui/ledmatrix.py:68]:
```suggestion
    while get_status() == 'random_eq':
```
More nitpicking :wink:

File: python/inputmodule/gui/ledmatrix.py [hunk 3]
```diff
@@ -72,7 +80,9 @@ def random_eq(dev):
 def clock(dev):
     """Render the current time and display.
     Loops forever, updating every second"""
-    while True:
+    animate(dev, False)
+    set_status('clock')
+    while True and get_status() == 'clock':
```
Comment (spenserblack) [inline python/inputmodule/gui/ledmatrix.py:85]:
```suggestion
    while get_status() == 'clock':
```
Even more nitpicking :sweat_smile:

File: python/inputmodule/inputmodule/ledmatrix.py [hunk 4]
```diff
@@ -120,7 +124,7 @@ def camera(dev):
         end_x = min(dim[1], start_x + WIDTH)
 
         # Pre-process the video into resized, cropped, grayscale frames
-        while True:
+        while True and get_status() == 'camera':
```
Comment (spenserblack) [inline python/inputmodule/inputmodule/ledmatrix.py:127]:
```suggestion
        while get_status() == 'camera':
```

File: python/inputmodule/inputmodule/ledmatrix.py [hunk 5]
```diff
@@ -161,7 +166,7 @@ def video(dev, video_file):
         processed = []
 
         # Pre-process the video into resized, cropped, grayscale frames
-        while True:
+        while True and get_status() == 'video':
```
Comment (spenserblack) [inline python/inputmodule/inputmodule/ledmatrix.py:169]:
```suggestion
        while get_status() == 'video':
```

File: python/inputmodule/inputmodule/ledmatrix.py [hunk 6]
```diff
@@ -294,8 +299,9 @@ def all_brightnesses(dev):
 def breathing(dev):
     """Animate breathing brightness.
     Keeps currently displayed grid"""
+    set_status('breathing')
     # Bright ranges appear similar, so we have to go through those faster
-    while True:
+    while True and get_status() == 'breathing':
```
Comment (spenserblack) [inline python/inputmodule/inputmodule/ledmatrix.py:304]:
```suggestion
    while get_status() == 'breathing':
```

**Comment by moon-jam** [comment]
Oops, I write a really bad code😅

**Review by JohnAZoidberg — APPROVED** [review]
Thanks! I cleaned up the commits

**Pull request merged**